invenio_records_permissions/permissions/base.py

```diff
@@ -6,99 +6,108 @@
 # and/or modify it under the terms of the MIT License; see LICENSE file for
 # more details.
 
+"""Base access controls."""
+
+from itertools import chain
+
 from flask import current_app
 from invenio_access import Permission
+
 from ..generators import Deny
 
+# Where can a property be used?
+#
+# |    Action   | need | excludes | query_filters |
+# |-------------|------|----------|---------------|
+# |    create   |   x  |     x    |               |
+# |-------------|------|----------|---------------|
+# |     list    |   x  |     x    |               |
+# |-------------|------|----------|---------------|
+# |     read    |   x  |     x    |        x      |
+# |-------------|------|----------|---------------|
+# | read files  |   x  |     x    |               |
```

Comment:
I beleive  in the long run search will be perforemed on files also so it should have a `query_filter` but we can just put a #TODO for now.